src/handleTask.js

```diff
@@ -0,0 +1,139 @@
+const taskList = document.getElementById('task-list');
+let editTaskDescription;
+let deleteTask;
+
+let tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+const saveTasks = () => {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+};
+
+const markAsCompleted = (task) => {
+  task.completed = true;
+};
+
+const markAsIncomplete = (task) => {
+  task.completed = false;
+};
+
+const createTaskLists = (task) => {
+  const deleteButton = document.createElement('button');
+  const listItemElement = document.createElement('li');
+  const iconElement = document.createElement('i');
+  const descriptionElement = document.createElement('span');
+
+  const checkboxElement = document.createElement('input');
+  checkboxElement.type = 'checkbox';
+  checkboxElement.checked = task.completed;
+
+  checkboxElement.addEventListener('change', () => {
+    if (checkboxElement.checked) {
+      markAsCompleted(task);
+    } else {
+      markAsIncomplete(task);
+    }
+    saveTasks();
+
+    // Check if the checkbox is now checked
+    if (checkboxElement.checked) {
+      deleteButton.style.display = 'block';
+      iconElement.style.display = 'none';
+      listItemElement.style.display = 'flex';
+      listItemElement.style.justifyContent = 'flex-start';
+      deleteButton.style.marginLeft = 'auto';
+    } else {
+      deleteButton.style.display = 'none';
+      iconElement.style.display = 'block';
+      descriptionElement.style.color = '#999';
+    }
+  });
+
+  descriptionElement.textContent = task.description;
+
+  descriptionElement.addEventListener('click', () => {
+    editTaskDescription(task);
+  });
+
+  listItemElement.appendChild(checkboxElement);
+  listItemElement.appendChild(descriptionElement);
+
+  iconElement.classList.add('fa', 'fa-ellipsis-v');
+  iconElement.addEventListener('click', () => {
+    editTaskDescription(task);
+    deleteButton.style.display = 'block';
+    iconElement.style.display = 'none';
+  });
+  listItemElement.appendChild(iconElement);
+
+  deleteButton.innerHTML = '<i class="fa fa-trash-o"></i>';
+  deleteButton.classList.add('delete-button');
+  deleteButton.style.display = 'none';
+
+  deleteButton.addEventListener('click', () => {
+    deleteTask(task.index);
+  });
+
+  listItemElement.appendChild(deleteButton);
+
+  return listItemElement;
+};
+
+function addNewTask(description) {
+  const taskIndex = tasks.length;
+
+  const task = { description, completed: false, index: taskIndex };
+  tasks.push(task);
+  saveTasks();
+
+  const listItemElement = createTaskLists(task);
+  taskList.appendChild(listItemElement);
+}
+
+const updateTaskIndexes = () => {
+  tasks.forEach((task, index) => {
+    task.index = index;
```

Comment:
- [ ] I will encourage you to add + 1 to this so that the index would start from 1, at the moment, it starts from 0 and the requirement says it should index from 1.

![image](https://github.com/Bhekisile/To-do-list/assets/86200356/7aa36043-e90f-4f1f-bc65-3cfea968efc6)

Reply:
Thank you for the feedback. I have made all the requested changes.